spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BiRecurrent.scala

```diff
@@ -135,10 +137,51 @@ class BiRecurrent[T : ClassTag] (
   }
 }
 
-object BiRecurrent {
+object BiRecurrent extends ContainerSerializable {
   def apply[@specialized(Float, Double) T: ClassTag](
     merge: AbstractModule[Table, Tensor[T], T] = null)
     (implicit ev: TensorNumeric[T]) : BiRecurrent[T] = {
     new BiRecurrent[T](merge)
   }
+
+  override def loadModule[T: ClassTag](model : BigDLModule)
+                                      (implicit ev: TensorNumeric[T]) : ModuleData[T] = {
+
+    val attrMap = model.getAttrMap
+
+    val merge = DataConverter.
+      getAttributeValue(attrMap.get("merge")).
+      asInstanceOf[AbstractModule[Table, Tensor[T], T]]
+
+    val biRecurrent = BiRecurrent(merge)
+
+    biRecurrent.birnn = DataConverter.
+      getAttributeValue(attrMap.get("birnn")).
+      asInstanceOf[Sequential[T]]
+
+    createBigDLModule(model, biRecurrent)
+  }
+
+  override def serializeModule[T: ClassTag](module : ModuleData[T])
+                                           (implicit ev: TensorNumeric[T]) : BigDLModule = {
+    val biRecurrentCls = Class.forName("com.intel.analytics.bigdl.nn.BiRecurrent")
+    val birecurrentModule = module.module.
+      asInstanceOf[BiRecurrent[T]]
+    val birecurrentBuilder = BigDLModule.newBuilder
+    birecurrentBuilder.setModuleType(biRecurrentCls.getName)
```

Comment:
yes, refined it. it used to be used for class registration, not used any more now